src/test/java/seedu/address/logic/commands/TagCommandIntegrationTest.java

```diff
@@ -0,0 +1,112 @@
+            CommandResult parserResult = tagCommand.execute(model);
+            Person defaultFirst = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+            Person editedPerson = new PersonBuilder(defaultFirst).withTags(VALID_TAG_FRIEND).build();
+            String expectedMessage = String.format(TagCommand.MESSAGE_TAG_ADD_SUCCESS,
+                    editedPerson.getName(), editedPerson.getTags());
+            CommandResult expectedResult = new CommandResult(expectedMessage);
+            assertEquals(expectedResult, parserResult);
+        } catch (ParseException e) {
+            assert false;
+        } catch (CommandException e) {
+            assert false;
+        }
+    }
+
+    @Test
+    public void execute_singleTagPreExistingTagUnfilteredList_success() {
+        Model model = generateDefaultTagModel();
+        Person defaultFirst = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person editedPerson = new PersonBuilder(defaultFirst).withTags(VALID_TAG_WIFE, VALID_TAG_FRIEND).build();
+
+        try {
+            TagCommand tagCommand = parser.parse(INDEX_FIRST_PERSON.getOneBased() + " " + PREFIX_TAG + VALID_TAG_WIFE
+                    + " " + PREFIX_TAG + VALID_TAG_FRIEND);
+            CommandResult parserResult = tagCommand.execute(model);
+
+            String expectedMessage = String.format(TagCommand.MESSAGE_TAG_ADD_SUCCESS,
+                    editedPerson.getName(), editedPerson.getTags());
+            CommandResult expectedResult = new CommandResult(expectedMessage);
+            assertEquals(expectedResult, parserResult);
+        } catch (ParseException e) {
+            assert false;
+        } catch (CommandException e) {
+            assert false;
+        }
+    }
+
+    @Test
+    public void execute_multipleTagPreExistingTagUnfilteredList_success() {
+        Model model = generateDefaultTagModel();
+        Person defaultFirst = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person editedPerson = new PersonBuilder(defaultFirst)
+                .withTags(VALID_TAG_WIFE, VALID_TAG_FRIEND, VALID_TAG_HUSBAND).build();
+        try {
+            TagCommand tagCommand = parser.parse(INDEX_FIRST_PERSON.getOneBased() + " " + PREFIX_TAG + VALID_TAG_WIFE
+                    + " " + PREFIX_TAG + VALID_TAG_FRIEND + " " + PREFIX_TAG + VALID_TAG_HUSBAND);
+            CommandResult parserCommandResult = tagCommand.execute(model);
+
+            String expectedMessage = String.format(TagCommand.MESSAGE_TAG_ADD_SUCCESS,
+                    editedPerson.getName(), editedPerson.getTags());
+
+            CommandResult expectedCommandResult = new CommandResult(expectedMessage);
+            assertEquals(expectedCommandResult, parserCommandResult);
+        } catch (ParseException e) {
+            assert false;
+        } catch (CommandException e) {
+            assert false;
+        }
+    }
+}
```

Comment:
I think the expected outcome can be more specific. If this method is testing command result would it be better to rename the expected outcome to a more specific phrase say, returnCommandResultEditPersonSuccess?